### ISSUE TITLE
Case that works with tag v0.3.3 dies with segmentation fault on d9d70e

### ISSUE THREAD

**Issue by jreniel** [dengwirda/jigsaw-python#18]
I have a case that runs OK on v0.3.3, but segfault on https://github.com/dengwirda/jigsaw-python/commit/d9d70e60fae9b5686c85113d1d1d4b21ae341789 with the error:
```
[...]
#------------------------------------------------------------

  Reading HFUN data...

**input error: HFUN. values must be non-negative.
```
However, when one inspects the Python objects passed to jigsawpy.libsaw.jigsaw they look like this:

```
np.min(hfun.value)=4.647238731384277                                                                                                                                                                                                         
np.max(hfun.value)=883.0133056640625                                                                                                                                                                                                         
opts.hfun_hmin=4.647238731384277                                                                                                                                                                                                             
opts.hfun_hmax=15000.0                                                                                                                                                                                                                       
```
An important note is that the same setup works in v0.3.3.

Note: The value of `opts.numthread` does not affect the outcome.

Thanks for looking into this!
-Jaime

**Comment by dengwirda** [comment]
Thanks for reporting this @jreniel. I haven't had luck reproducing it yet though --- would you mind exporting the various `*.msh` and `*.jig` files and attaching here?

**Comment by jreniel** [comment]
I will followup with a link to a set of files in which the problem shows up soon. It may take a me a few days while I sort out a different set of issues on my end.
In the meantime, one clue that might help is that my hfun is of  euclidean-mesh type, not `euclidean-grid` (it's actually a regular grid with "split-quads").  The reason is that to reproject a regular grid from geographic coordinates to local coordinates, one cannot do it with an euclidean-grid without re-interpolating. Converting the regular to an unstructured grid allows me to reproject the grid without reinterpolating. Also it didn't happen with every case, only some ... Anyhow, I will follow-up soon with some files.
Thanks!
-J.

**Comment by jreniel** [comment]
I just found a case where it was segfaulting, and I wrote the files with savemsh/jig ... Then I ran the jigsaw binary on them directly and it didn't segfault. I think the problem is coming from passing the hfun as a pointer to libsaw. Note that on this latest example, my hfun has 9 million nodes. This must be related to some overflow due to larger than expected array.
I have the files saved, the hfun.msh is 631Mb. I will see if I can give you a link to somewhere.
But these pointers should give you clues about what's happening. The last version I tested that segfaulted was da9e25624828f83de56b0f7d81aef4b9c75114b4 . I would need to continue gradually rolling back versions until I get to one that doesn't segfault. But like I said, I think the problem happens only when passing large hfun arrays (eiclidean-mesh) to libsaw from ctypes..
